### ISSUE TITLE
coverage analysis should take into account enumerated bounds

### ISSUE THREAD

**Issue by gavinking** [ceylon/ceylon-spec#1278]
The typechecker should accept this code:

```ceylon
T fun<T>(T x, T y) 
        given T of Float|Integer {
    switch (pair = [x,y])
    case (is [Integer,Integer]) { return pair[0]+pair[1]; }
    case (is [Float,Float]) { return pair[0]+pair[1]; }
}
```

I'm not sure how hard this is.

Related: #560.

**Comment by RossTate** [comment]
You're reified, so I don't get why you don't just switch on `T`.

**Comment by gavinking** [comment]
@RossTate well, sure, that's essentially what #560 is on about.

**Comment by RossTate** [comment]
Well, here's how you could go about typechecking this and other things like it: type check it for each `of` case on `T`. If you think that's a bad idea, then I'd say the hacks that would be necessary to make this type check are probably a worse idea. So I'd recommend dropping either extreme but not the middle ground.

**Comment by gavinking** [comment]
@RossTate Yes, that approach had already occurred to me. You need to be a little bit careful or the compiler will output dupe errors, but in principle it could work.
